src/transformStateWithClones.js

```diff
@@ -6,8 +6,37 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
   // write code here
+  const array = [];
+  let stateCopy = { ...state };
+  let pushedCopy = {};
+
+  for (const object of actions) {
+    switch (object.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, object.extraData);
+        break;
+
+      case 'removeProperties':
+        object.keysToRemove.forEach(Element => {
+          delete stateCopy[Element];
+        });
+        break;
+
+      case 'clear':
+        stateCopy = {};
```

Comment:
Now, since the stateCopy is a const, you can get back to a loop for deleting properties

Rembember to check `hasOwnProperty` before deleting it!

Reply:
I put hasOwnProperty on deleting chosen keys. On clear it's not necessary, becouse we're removing ALL the keys.